docs/in-app-purchase-validation.md

```diff
@@ -1 +1,207 @@
+- ...and more, with new vulnerabilities emerging occasionally.
+
+For in-app purchases, a trusted source of truth is required. Nakama checks and tracks purchases and purchase history, solving a significant set of possible vulnerabilities and pain points.
+
+In-App Purchase Validation is available for Apple and Google purchases, regardless of platform. Both single product and subscription purchases are supported.
+
+__Fake Purchases__
+
+Nakama directly connects to Apple and Google services to check the validity of all incoming payments. This verification is completely outside the client's code, and cannot be intercepted and tampered with.
+
+Every transaction is verified, every time, and invalid ones are rejected.
+
+__Replay Attacks__
+
+All transactions are logged, preventing multiple submissions of the same purchase token or receipt.
+
+__Receipt Sharing__
+
+Successful transactions are bound to the account that submits them. Different users cannot submit the same transaction, even a valid one, in an attempt to receive the associated reward.
+
+__Product Mismatches__
+
+The transaction is checked to ensure the correct reward is tied to each purchase. This prevents attacks that attempt to use a valid (cheap) purchase to unlock a different (expensive) reward.
+
+__Subscription Expiry__
+
+Nakama checks subscriptions to see if they've expired, and rejects the transaction as needed.
+
+__Purchase Cancellation__
+
+Valid receipts that link to cancelled purchases are flagged and rejected.
+
+__Single Source of Truth__
+
+While Nakama maintains an internal record of all transactions, the remote payment provider is always given priority. Valid purchases that have been checked, logged, then subsequently cancelled, will be rejected appropriately.
+
+## Apple
+
+Nakama supports validating purchases made for products and subscription in iOS.
+
+Apple purchase receipts are sent to Apple for validation. As suggested by Apple, both Production and Sandbox servers are used to validate receipts depending on the priority setup in the Nakama configuration.
+
+### Setup
+
+To validate receipts against the App Store, Nakama requires your app's shared secret. You can setup a shared secret in [iTunes Connect](https://itunesconnect.apple.com).
+
+![Apple iTunes Connect][apple_iap_1]
+
+Make a record of your shared secret:
+
+![Apple iTunes Connect Shared Secret][apple_iap_2]
+
+You'll need to set the value of `purchase.apple.password` to the value of the Shared Secret above. For more info, take a look at the [configuration](configuration.md#purchase) page.
+
+If your app is in production, you'll need to set the value of `purchase.apple.production` to true to give priority Apple's Production servers.
+
+### Validate Purchase
+
+Nakama only supports validating iOS 7+ receipts. In addition, Nakama only validates the first item in the receipt as Apple receipts can contain more than one in-app purchase item.
```

Comment:
Is there a way here for us to clarify/offer a solution to dev's to not allow players to buy multiple IAP's in order to prevent the multiple IAP's on one receipt?